nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/controllers/LabeledFieldController.java

```diff
@@ -56,7 +73,26 @@ public String getLabel() {
      *                  field can be empty.
      */
     public void setIsRequired(boolean required) {
-        this.required = required;
+        if (! required) {
+            // Multiple instances of the validator may be present.
+            Iterator<InputValidator> iterator = validators.iterator();
+            while (iterator.hasNext()){
+                if(iterator.next() instanceof RequiredFieldValidator) {
+                    iterator.remove();
+                }
+            }
+        } else if (! isRequired()) {
+            validators.add(new RequiredFieldValidator());
```

Comment:
It would require to implement either the `equals/hashCode` or the `singleton` method to enable the user to see a duplicate error message at runtime, if I understood well what you intend to do. 

From what I understand from your last message we would only keep a particular instance of `RequiredFieldValidator` in `LabeledItemController` and let the user create a `RequiredFieldValidator` if he wants to set it with his own validators? (without any identification method stated above)

I don't quite see the advantage since it keeps the idea of a singleton but limited to `LabeledItemController`, why not providing the instance to the user then?
Moreover, the use of one of the two methods described above will simplify the implementation of `isRequired` to a single `validators.contains(...)` call